07-Threads/threads.c

```diff
@@ -49,23 +49,37 @@ void systick_handler()
 	*SCB_ICSR |= SCB_ICSR_PENDSVSET;
 }
 
-void thread_start()
+void __attribute__((naked)) thread_start()
 {
 	lastTask = 0;
 
-	/* Save kernel context */
-	asm volatile("mrs ip, psr\n"
-	             "push {r4-r11, ip, lr}\n");
-
+	/* Reset APSR before context switch */
```

Comment:
You shall explain the reason.

Reply:
Actually, I am hesitate to reset this. This in fact has no influence in switching first task unless the task use condition flag instruction right away switching to it.

It's just like there is a `int *p`, and I want to initialize it to `int *p = NULL`.

This might also complicate the flow, even though we had assigned stack[16] as reset value of `psr` and should also keep consistent here with `pendsv_handler`.

What do you think about this?